core/ArchiveProcessor/Loader.php

```diff
@@ -73,12 +77,21 @@ private function prepareArchiveImpl($pluginName)
             return $idArchive;
         }
 
-        list($visits, $visitsConverted) = $this->prepareCoreMetricsArchive($visits, $visitsConverted);
-        list($idArchive, $visits) = $this->prepareAllPluginsArchive($visits, $visitsConverted);
+        /** @var ArchivingStatus $archivingStatus */
+        $archivingStatus = StaticContainer::get(ArchivingStatus::class);
+        $lock = $archivingStatus->archiveStarted($this->params);
 
-        if ($this->isThereSomeVisits($visits) || PluginsArchiver::doesAnyPluginArchiveWithoutVisits()) {
-            return $idArchive;
+        try {
+            list($visits, $visitsConverted) = $this->prepareCoreMetricsArchive($visits, $visitsConverted);
+            list($idArchive, $visits) = $this->prepareAllPluginsArchive($visits, $visitsConverted);
+
+            if ($this->isThereSomeVisits($visits) || PluginsArchiver::doesAnyPluginArchiveWithoutVisits()) {
+                return $idArchive;
+            }
+        } finally {
+            $archivingStatus->archiveFinished($lock);
```

Comment:
fyi `archiveFinished ` does not take an argument  and above the `archiveStarted ()` does not return a value

Reply:
fixed